pkg/controller/common/util.go

```diff
@@ -603,6 +603,12 @@ func GetPodsUsingPVCs(ctx context.Context, c client.Client, namespace string, na
 									onlyReadOnly = false
 								}
 							}
+							for _, vm := range c.VolumeDevices {
+								if vm.Name == volume.Name {
+									// Node level rw mount and container can't mount block device ro
+									onlyReadOnly = false
+								}
+							}
```

Comment:
@mhenriks @awels I think this check was always broken for pods with volumeDevices

Reply:
Kind of.  This will only cover cases where block PVCs are in the pod spec read/write but not referred to by a container.